src/workers/postUpdated.ts

```diff
@@ -75,76 +292,49 @@ const worker: Worker = {
           return;
         }
 
-        const creatorTwitter =
-          data?.extra?.creator_twitter === '' ||
-          data?.extra?.creator_twitter === '@'
-            ? null
-            : data?.extra?.creator_twitter;
-
-        const authorId = await findAuthor(entityManager, creatorTwitter);
-        const title = data?.title || databasePost.title;
-        const becomesVisible = !!title?.length;
-
-        const { allowedKeywords, mergedKeywords } = await mergeKeywords(
+        const { mergedKeywords, fixedData } = await fixData({
+          logger,
           entityManager,
-          data?.extra?.keywords,
-        );
+          data,
+        });
 
-        if (allowedKeywords.length > 5) {
-          logger.info(
+        // See if post id is not available
+        let combinedPost: Partial<ArticlePost>;
+        if (!post_id) {
+          // Handle creation of new post
+          try {
+            combinedPost = await createPost({
+              logger,
+              entityManager,
+              data: fixedData,
+              submission_id: data?.submission_id,
+            });
+          } catch (e) {
+            return;
+          }
+        } else {
+          // Handle update of existing post
+          combinedPost = await updatePost({
+            entityManager,
+            data: fixedData,
+            id: post_id,
+          });
+        }
+
+        // After add or update:
+        if (combinedPost.visible) {
```

Comment:
Should we always do this check? I think we should do it only on creation of if `visible` was changed right?

Reply:
Hmm, yeah originally `newPost` would check for title atleast, in this new flow it's generic so there could be a case where it's not yet.
Not sure if we could ever create new post that have share post attached actually.

Reply:
Oh wait you are right, let me adjust it back